crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -613,21 +606,19 @@ fn convert_onchain_order_placement(
         false
     };
 
-    let liquidity_owner = if liquidity_order_owners.contains(&owner) {
+    // remove liquidity owners as part of the https://github.com/cowprotocol/infrastructure/issues/1032
+    // leaving uncommmented for compiler to catch
+    let _liquidity_owner = if liquidity_order_owners.contains(&owner) {
         tracing::debug!(%order_uid, ?owner, "order being flagged as placed by liquidity order owner");
         true
     } else {
         false
     };
 
-    // TODO(nlordell): It is currently possible to create limit orders from
-    // on-chain events even if they are disabled at the API level. This feels
-    // non-intentional, and we should revisit this before releasing EthFlow
-    // orders.
-    let class = match (is_outside_market_price, liquidity_owner) {
-        (true, true) => OrderClass::Liquidity,
-        (true, false) => OrderClass::Limit,
-        _ => OrderClass::Market,
+    // For market orders, fee is charged from the signed fee amount.
+    let class = match (order_data.fee_amount.is_zero(), is_outside_market_price) {
+        (false, false) => OrderClass::Market,
+        _ => OrderClass::Limit,
```

Comment:
Main change of the PR

Reply:
I know this was the logic before, but do we even support `LimitOrders` with non-zero fee (if I understand solvers cannot add a fee to those)?

What happens now if the user sends a non-zero fee order that is out of market? It gets classified as Limit and then what? Can we add an e2e test for this behavior (my gut feeling is that we should not allow those orders if they cause issues)?

Reply:
Yes, we do not support limit orders with user fee.

I can completely remove the `is_outside_market_price` and rely only on the `fee_amount` to determine the class (which would align it with how the class is determined in the orderbook). Then, we are safe.

Whether quote should be used to classify the order and when we should switch to this, is outside of the scope of this PR i'd say and should be part of cleanup of OrderClass::Market variant.

Reply:
I think we didn't want to allow users to place market orders with insufficient since the protocol would eat the loss (ie. now someone could place an order with 1wei fee and practically pay no fee). Those orders should be rejected somehow IMO.